### DISCUSSION TITLE
Encoding bitmaps to mp4 sample

### DISCUSSION THREAD

**Discussion by goodintention** [FactoryXCode/MfPack#45] [Ideas]
Can you please add a sample for encoding bitmaps to mp4 video?
Thanks.

**Comment by FactoryXCode** [comment]
Hi, Thanks for your question.
For this, you have to explain what you want to accomplish, and for example a raw data flow/schedule and interfaces to use.
Regards Tony.

**Comment by goodintention** [comment]
Thanks for the reply,
I found a way to encode bitmaps to mp4 by encoding to AVI then transcoding to MP4
also, I found a way using the sink writer, but it's hard for me to translate it to Delphi 
https://learn.microsoft.com/en-us/windows/win32/medfound/tutorial--using-the-sink-writer-to-encode-video?redirectedfrom=MSDN

**Comment by FactoryXCode** [comment]
Ok, I translated the cpp code.
I dropped some working samples [here](https://github.com/FactoryXCode/MfPack/tree/Master/MfPack/Samples/SinkWriterToEncodeVideoSample).

Please update MfApi and MfReadWrite first to latest version. 

Regards, Tony.

**Reply by goodintention** [reply]
The sample is quite helpful. Many thanks, Tony.
although I can't find the "mark as answer" button!

**Reply by FactoryXCode** [reply]
You're welcome.
The sample is updated to support MP4, AVI and WMV.
Also, Sample2 shows how to render a bitmapfile in to a videofile like MP4, AVI or WMV .
